### ISSUE TITLE
Add functionality to punish agents

### ISSUE THREAD

**Issue by dhruvds12** [ADimoska/SOMASExtended#48]
As mentioned in lectures and meeting, there needs to be functionality to punish agents that don't follow the AoA. I think this should be a function that can be added to the AoA interface, and leave it to each team to implement themselves. 

This will require another voting stage after the audit has been completed so that agents can choose to enact a punishment on the "cheating" agent.

**Comment by arnavxkohli** [comment]
I had this in the original AoA, but I think this shouldn't be in the interface, because the function signature may be different across different teams. See https://github.com/ADimoska/SOMASExtended/pull/30

**Comment by johanjino** [comment]
The server surely wont know what punishment to enforce, only the AoA does? Wouldnt that need a function in the AoA

**Comment by arnavxkohli** [comment]
The server can call a switch case depending on which specific AoA is being called. Let's say I want my punishment to be that the other agents decide if I roll or not. This cannot be a function within the AoA, because the server is the one calling the agents for their roll or stick decision (correct me if I'm wrong).

**Comment by arnavxkohli** [comment]
It can be an unimplemented field within the AoA, so feel free to make the change if you want to

**Comment by johanjino** [comment]
Got it, I understand that AoA might want to change some specific functionality for some agents as punishment. 

Although, are we planning on handling these for any type of punishment any team wants? This would mean a lot of changes on server from each AoA.

**Comment by arnavxkohli** [comment]
It would but I think we can leave that to be the teams' individual responsibility. If they aren't able to code it and make it work with everyone else's code we can ask them to change their AoA or something else

**Comment by johanjino** [comment]
Okay makes sense, but lets do that after MVP? Since this changes lots of stuff on the server, which can cause issues for everyone.

**Comment by dhruvds12** [comment]
Sounds good, I agree with Johan, let's keep it simple for MVP. Happy for another issue to be raised for a more complicated punishment implementation to be implemented under the Extension milestone.

**Comment by dhruvds12** [comment]
Has this been implemented or do we still need to do this?

@johanjino @arnavxkohli

**Comment by arnavxkohli** [comment]
It's pending for team 2. Currently waiting on kicking agents from the team. Will probably just implement it myself now. Only thing I'm a little iffy about is the orphan pool management. @dhruvds12
